programs/drift_vaults/src/state/vault_depositor.rs

```diff
@@ -501,14 +512,74 @@ impl VaultDepositor {
             total_vault_shares_after: vault.total_shares,
             user_vault_shares_after: vault.user_shares,
             profit_share,
-            management_fee: 0,
+            management_fee,
         });
 
-        let user_withdraw_amount = withdraw_amount.safe_sub(profit_share.cast()?)?;
         vault.total_withdraw_requested =
             vault.total_withdraw_requested.safe_sub(withdraw_amount)?;
 
-        Ok(user_withdraw_amount)
+        Ok(withdraw_amount)
+    }
+
+    pub fn apply_profit_share(
+        self: &mut VaultDepositor,
+        vault_equity: u64,
+        vault: &mut Vault,
+    ) -> Result<u64> {
+        let n_shares = self.vault_shares;
+        let amount: u64 =
+            depositor_shares_to_vault_amount(n_shares, vault.total_shares, vault_equity)?;
+
+        let total_amount =
+            depositor_shares_to_vault_amount(self.vault_shares, vault.total_shares, vault_equity)?;
```

Comment:
think you want vault.total_shares here

Reply:
or just vault_equity

Reply:
simplified this since its no longer based on realized fraction but just fully applied to profits